packages/tailwindcss/src/variants.ts

```diff
@@ -898,32 +898,35 @@ export function createVariants(theme: Theme): Variants {
   return variants
 }
 
-function quoteAttributeValue(value: string) {
-  if (value.includes('=')) {
-    value = value.replace(/(=.*)/g, (_fullMatch, match) => {
-      // If the value is already quoted, skip.
-      if (match[1] === "'" || match[1] === '"') {
-        return match
-      }
+function quoteAttributeValue(input: string) {
+  if (input.includes('=')) {
+    let [attribute, ...after] = segment(input, '=')
+    let value = after.join('=')
+    let valueTrimmed = value.trim()
+
+    // If the value is already quoted, skip.
+    if (valueTrimmed[0] === "'" || valueTrimmed[0] === '"') {
+      return input
+    }
 
-      // Handle regex flags on unescaped values
-      if (match.length > 2) {
-        let trailingCharacter = match[match.length - 1]
-        if (
-          match[match.length - 2] === ' ' &&
-          (trailingCharacter === 'i' ||
-            trailingCharacter === 'I' ||
-            trailingCharacter === 's' ||
-            trailingCharacter === 'S')
-        ) {
-          return `="${match.slice(1, -2)}" ${match[match.length - 1]}`
-        }
+    // Handle regex flags on unescaped values
```

Comment:
I know this isn't new to this PR but this comment confused me:

```suggestion
    // Handle case sensitivity flags on unescaped values
```